### ISSUE TITLE
contour plot error using eclipse launch script

### ISSUE THREAD

**Issue by lizadams** [CEMPD/VERDI#101]
I am getting the following error when I try to create a contour plot within Eclipse of the O3 variable for the CCTM46_P16.base02a.36k.O3MAX file.
The error message is:
Variable indexToVertexID not found, not MPAS
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: javax/media/opengl/GLCapabilitiesChooser
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
    at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
    at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:264)
    at javax.media.j3d.Pipeline$1.run(Pipeline.java:162)
    at java.security.AccessController.doPrivileged(Native Method)
    at javax.media.j3d.Pipeline.createPipeline(Pipeline.java:157)
    at javax.media.j3d.MasterControl.loadLibraries(MasterControl.java:965)
    at javax.media.j3d.VirtualUniverse.<clinit>(VirtualUniverse.java:299)
    at javax.media.j3d.Canvas3D.<clinit>(Canvas3D.java:3881)
    at anl.verdi.plot.types.Contour3D.<init>(Contour3D.java:174)
    at anl.verdi.plot.gui.PlotFactory.getPlot(PlotFactory.java:95)
    at anl.verdi.plot.gui.PlotFactory.getPlot(PlotFactory.java:78)
    at anl.verdi.plot.gui.DefaultPlotCreator.doCreatePlot(DefaultPlotCreator.java:48)
    at anl.verdi.plot.gui.AbstractPlotCreator.createPlot(AbstractPlotCreator.java:39)
    at anl.verdi.plot.gui.action.Contour3DPlot.actionPerformed(Contour3DPlot.java:22)
    at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
    at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
    at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
    at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
    at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
    at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
    at java.awt.Component.processMouseEvent(Component.java:6535)
    at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
    at java.awt.Component.processEvent(Component.java:6300)
    at java.awt.Container.processEvent(Container.java:2236)
    at java.awt.Component.dispatchEventImpl(Component.java:4891)
    at java.awt.Container.dispatchEventImpl(Container.java:2294)
    at java.awt.Component.dispatchEvent(Component.java:4713)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
    at java.awt.Container.dispatchEventImpl(Container.java:2280)
    at java.awt.Window.dispatchEventImpl(Window.java:2750)
    at java.awt.Component.dispatchEvent(Component.java:4713)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:709)
    at java.awt.EventQueue$3.run(EventQueue.java:703)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
    at java.awt.EventQueue$4.run(EventQueue.java:731)
    at java.awt.EventQueue$4.run(EventQueue.java:729)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.ClassNotFoundException: javax.media.opengl.GLCapabilitiesChooser
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 63 more

**Comment by lizadams** [comment]
fixed this error by adding the following jar files to the classpath for the verdi_script_contour launch script:
verdi_core/lib/macosx/
jogl-all-natives-macosx-universal.jar
j3dutils.jar
j3dcore.jar
gluegen-rt.jar
gluegen-rt-natives-macosx-universal.jar
vecmath.jar
jogl-all.jar

**Comment by systemsgo** [comment]
I see a couple of complications with this approach.  Modifying the launch script means each script would have to be modified to enable the contour plot.  Also there are native libraries, some of which are in platform specific directories, so the scripts won't work across platforms.  We could move all the native libraries to a common directory, but there's still another problem - we're including Java3D 1.6 for MacOS, but version 1.5.2 for Windows and Linux.  Some of the jars have the same names, so we'd get collisions if we tried to have both versions in the same path / classpath.  I would upgrade everything to 1.6, but I can't find a 1.6 build for 64 bit Windows.  Is there a need for 1.6 on MacOS?  If not we could revert to 1.5.2, and everything would work pretty cleanly.

I checked in changes to verdi_core and verdi_bootstrap's .classpath files that should let all of the run configurations work on Linux and Windows.  The Linux ones work, while the windows ones are failing due to what look like OpenGL driver issues in my Windows VM.  The fact that it's even trying makes me think it's loading the libraries correctly though.

I doubt my changes will work for the Mac version because of the mix of Java3D 1.5.2/1.6.  If we can standardize on a single version we should be able to make everything work without needing to change the launch scripts.
